greeter/README.md

```diff
@@ -1,3 +1,121 @@
+
+In this workshop, you will be guided to scaffold a greeter app using the LoopBack 4 framework, and create a simple greeting API that prints out a welcome message, then enrich it without modifying its implementation in the next 4 steps:
+
+1. The greeting service is packaged as a separate module from the greeting app (modular)
+2. The greeting API can speak different languages based on the client locale. The support for new languages can be plugged in (extensibility and pluggability)
+3. Enable the caching with sweeping for REST requests (interception/observation)
+4. The greeting app and certain greeters have configuration options (configurability)
+
+_TODO: Add diagram. Maybe similar to https://github.com/strongloop/loopback-next/blob/master/examples/greeting-app/greeting-app.png_
+
+This miniature project reflects many key aspects of a scalable and extensible applications, and relevant techniques will be covered in the workshop, like Inversion of Control, Dependency Injection, Decorators and Component.
+
+
+## Before We Begin
+To complete the steps in this tutorial, you need to install Node.js and the LoopBack 4 command line interface (CLI).
+
+1. Install Node.js version 8.9 or higher. 
+    See the [Node documentation](https://nodejs.org/en/download/) for installation instructions.
+    
+    Tip: We recommend to use [Node Version Manager (nvm)](https://github.com/nvm-sh/nvm) to make it easier to manage the Node.js versions you're using. 
+
+2. Install LoopBack 4 CLI
+
+    The LoopBack 4 CLI is a command-line interface that scaffolds an application as well as artifacts (for example, controllers, models, and repositories) for existing applications. The CLI provides the fastest and easiest way to get started with a LoopBack 4 project that adheres to best practices.
+
+    Install the CLI globally by running the following command:
+
+    ```sh
+    npm i -g @loopback/cli
+    ```
+
+## Part 1: Create a Greeting Service
+
+### Step 1: Create a Message Type File
+We're going to create a `Message` type. In the `src` folder, create a file called `types.ts` with the following content:
+
+```ts
+/**
+ * Greeting message response
+ */
+export interface Message {
+  timestamp: Date;
+  language: string;
+  greeting: string;
+}
+```
+
+### Step 2: Create a Greeting Controller
+
+Create an endpoint `/greet/{name}` that takes a name and returns a greeting message in the language that is specified in the HTTP request. 
+
+```sh
+$ lb4 controller
+? Controller class name: Greeting
+Controller Greeting will be created in src/controllers/greeting.controller.ts
+
+? What kind of controller would you like to generate? Empty Controller
+   create src/controllers/greeting.controller.ts
+   update src/controllers/index.ts
+
+Controller Greeting was created in src/controllers/
+```
+
+- modify the constructor
```

Comment:
There are a few more trivial things to add (like the mock/simplified greeter service)

Reply:
I saw the simplified greeter service in your instruction too https://github.com/strongloop/cascon2019/blob/master/greeter/src/controllers/greeting.controller.ts#L10-L15.  but wondering why we cannot just use the one from the greeter-extension. :)